src/Proto.Actor/PID.cs

```diff
@@ -40,44 +40,18 @@ public void Request(object message, PID sender)
 
 
         public Task<T> RequestAsync<T>(object message, TimeSpan timeout)
-            => RequestAsync<T>(message, new CancellationTokenSource(timeout));
+            => RequestAsync(message, new FutureProcess<T>(timeout));
 
         public Task<T> RequestAsync<T>(object message, CancellationToken cancellationToken)
-            => RequestAsync<T>(message, CancellationTokenSource.CreateLinkedTokenSource(cancellationToken));
+            => RequestAsync(message, new FutureProcess<T>(cancellationToken));
 
         public Task<T> RequestAsync<T>(object message)
-            => RequestAsync<T>(message, null);
+            => RequestAsync(message, new FutureProcess<T>());
 
-        private Task<T> RequestAsync<T>(object message, CancellationTokenSource cts = null)
+        private Task<T> RequestAsync<T>(object message, FutureProcess<T> future)
         {
-            var tsc = new TaskCompletionSource<T>();
-
-            var reff = (cts == null) ? new FutureProcess<T>(tsc) : new FutureProcess<T>(tsc, cts.Token);
-
-            var name = ProcessRegistry.Instance.NextId();
-            var (pid, absent) = ProcessRegistry.Instance.TryAdd(name, reff);
-            if (!absent)
-            {
-                throw new ProcessNameExistException(name);
-            }
-            Request(message, pid);
-
-
-            if (cts == null) return tsc.Task;
```

Comment:
I think it would be better to keep everything above this line in PID, and only move the Task mojo into FutureProcess. Like this:
```csharp
var tsc = new TaskCompletionSource<T>();
var reff = (cts == null) ? new FutureProcess<T>(tsc) : new FutureProcess<T>(tsc, cts);		
var name = ProcessRegistry.Instance.NextId();		
var (pid, absent) = ProcessRegistry.Instance.TryAdd(name, reff);		
if (!absent)		
{		
    throw new ProcessNameExistException(name);		
}		
Request(message, pid);
return reff.Task;
```